heron/scheduler-core/src/java/com/twitter/heron/scheduler/utils/SchedulerUtils.java

```diff
@@ -39,12 +40,105 @@
 import com.twitter.heron.spi.utils.ShellUtils;
 
 public final class SchedulerUtils {
-  public static final int PORTS_REQUIRED_FOR_EXECUTOR = 9;
+  public static final int PORTS_REQUIRED_FOR_EXECUTOR = ExecutorPorts.getRequiredNumOfPorts();
   public static final int PORTS_REQUIRED_FOR_SCHEDULER = 1;
   public static final String SCHEDULER_COMMAND_LINE_PROPERTIES_OVERRIDE_OPTION = "P";
 
   private static final Logger LOG = Logger.getLogger(SchedulerUtils.class.getName());
 
+  /**
+   * Enum that defines the type of ports that an heron executor needs
+   */
+  public static class ExecutorPorts extends HashMap<ExecutorPorts.ExecutorPortNames, Object> {
+    private static final long serialVersionUID = 3594142561871208879L;
+
+    public enum ExecutorPortNames {
```

Comment:
This enum of port names provides little more than a list of strings. I suggest instead having an enum of ports (not names) and include metadata of each, like whether it's required or not and it's default value. Then enum itself could then have a getter for fetching just the required ones for example.

One benefit of the proposed approach is that to add a new port you have to modify the enum and explicitly set whether it's required or not when doing it. As written someone needs to add a port name and know that there's also another place in code to modify to represent required or not.

Also enum names should be singular not plural. That way you work with a singular enum (e.g., `doSomething(ExecutorPortName name)`) or the with all of them (e.g. `ExecutorPortName[] all = ExecutorPortName.values()`).